hs_core/hydroshare/hs_bagit.py

```diff
@@ -203,4 +211,39 @@ def create_bag(resource):
         timestamp=resource.updated
     )
 
-    return b
\ No newline at end of file
+    return b
+
+def read_bag(bag_path):
+    """
+
+    :param bag_path:
+    :return:
+    """
+    #import pdb; pdb.set_trace()
+    tmpdir = None
+    unpacked_bag_path = None
+
+    try:
+        if not os.path.exists(bag_path):
+            raise HsBagitException('Bag does not exist')
+        if os.path.isdir(bag_path):
+            unpacked_bag_path = bag_path
+        else:
+            mtype = mimetypes.guess_type(bag_path)
+            if mtype[0] != 'application/zip':
+                msg = "Expected bag to have MIME type application/zip, but it has {0} instead.".format(mtype[0])
+                raise HsBagitException(msg)
+            tmpdir = tempfile.mkdtemp()
+            zfile = zipfile.ZipFile(bag_path)
+            zroot = zfile.namelist()[0].split(os.sep)[0]
+            zfile.extractall(tmpdir)
+            unpacked_bag_path = os.path.join(tmpdir, zroot)
+
+        bag = bagit.Bag(unpacked_bag_path)
+        if not bag.is_valid():
+            msg = "Bag is not valid"
+            raise HsBagitException(msg)
+
+    finally:
+        if tmpdir:
+            shutil.rmtree(tmpdir)
```

Comment:
This function is currently not used, is incomplete, and should not be used.  This functional will be completed and used once this branch is merged and used to allow migration of old resources to the new base class and access control system.